src/wolfsftp.c

```diff
@@ -2626,13 +2655,16 @@ static int wolfSSH_SFTPNAME_readdir(WOLFSSH* ssh, WDIR* dir, WS_SFTPNAME* out,
         char  r[WOLFSSH_MAX_FILENAME];
         char  s[WOLFSSH_MAX_FILENAME];
 
-        r[0] = '\0';
-        WSTRNCAT(r, dirName, sizeof(r));
-        WSTRNCAT(r, "/", sizeof(r));
-        WSTRNCAT(r, out->fName, sizeof(r));
-        r[sizeof(r) - 1] = 0;
+        if ((WSTRLEN(dirName) + WSTRLEN(out->fName) + 2) > sizeof(r)) {
+            WLOG(WS_LOG_SFTP, "Path length too large");
```

Comment:
Not sure how we intend to use WS_LOG_SFTP but am curious if this and other error logs should use WS_LOG_ERROR instead.